source/includes/library-reference/_simulator.html.md

```diff
@@ -0,0 +1,158 @@
+# Simulator Class
+
+> ![Simulator state image][1]
+
+This class is used to interface with the server while training or running predictions against
+a **BRAIN**. It is an abstract base class, and the developer must create a subclass.
+
+The `Simulator` class has a close relationship with the **Inkling** file associated with
+the **BRAIN**. The name used to construct the `Simulator` must match the name of the simulator
+in the *Inkling* file.
+
+There are two main methods one must override, `episode_start` and `simulate`.
+
+## Brain brain()
```

Comment:
It's unusual to see return types in Python documentation.  I think a good style to emulate is SQL Alchemy: http://docs.sqlalchemy.org/en/latest/core/connections.html#sqlalchemy.engine.Engine.  There is a styleguide for Python documentation: https://devguide.python.org/documenting/ but it assumes use of sphinx and restructured text.

Reply:
C++